pkg/api/v1alpha1/gitopsconfig_types.go

```diff
@@ -86,6 +86,34 @@ func (c *GitOpsConfig) ExpectedKind() string {
 	return GitOpsConfigKind
 }
 
+func (c *GitOpsConfig) ConvertToFluxConfig() *FluxConfig {
+	if c == nil {
+		return nil
+	}
+	config := &FluxConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       FluxConfigKind,
+			APIVersion: c.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.Name,
+			Namespace: c.Namespace,
+		},
+		Spec: FluxConfigSpec{
+			SystemNamespace:   c.Spec.Flux.Github.FluxSystemNamespace,
+			Branch:            c.Spec.Flux.Github.Branch,
+			ClusterConfigPath: c.Spec.Flux.Github.ClusterConfigPath,
+			Github: &GithubProviderConfig{
+				Owner:      c.Spec.Flux.Github.Owner,
+				Repository: c.Spec.Flux.Github.Repository,
+				Personal:   c.Spec.Flux.Github.Personal,
+			},
+		},
+	}
+	config.ObjectMeta.Name = c.Name
```

Comment:
why do we need this?

Reply:
good catch, I can remove that -- in my first pass at making this work, I converted the type and then used the `FluxConfig` type on the cluster as well instead of marshalling the GitOpsConfig, this is left over from that. Will remove.

Reply:
https://github.com/aws/eks-anywhere/pull/1827/commits/ee9424dc17ca74abb8f530dc7c1fb36d1373f901